### PR TITLE
spring.resources.cache.cachecontrol is configured in webFluxAutoConfiguration

### PR DESCRIPTION
**Pull request by ayudovin**
this [bug](https://github.com/spring-projects/spring-boot/issues/14724)

### CODE REVIEW

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration.java [hunk 1]
```diff
@@ -149,14 +149,16 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
...
+			CacheControl cacheControl = this.resourceProperties.getCache()
+					.getCachecontrol().toHttpCacheControl();
 			if (!registry.hasMappingForPattern("/webjars/**")) {
 				ResourceHandlerRegistration registration = registry
 						.addResourceHandler("/webjars/**")
 						.addResourceLocations("classpath:/META-INF/resources/webjars/");
 				if (cachePeriod != null) {
-					registration.setCacheControl(CacheControl
-							.maxAge(cachePeriod.toMillis(), TimeUnit.MILLISECONDS));
+					cacheControl.sMaxAge(cachePeriod.toMillis(), TimeUnit.MILLISECONDS);
```
Comment (bclozel) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration.java]: `sMaxAge` looks like a typo here - if the current tests are not breaking with that, they're probably not testing it right.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration.java [hunk 2]
```diff
@@ -165,9 +167,9 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
 						.addResourceHandler(staticPathPattern).addResourceLocations(
 								this.resourceProperties.getStaticLocations());
 				if (cachePeriod != null) {
-					registration.setCacheControl(CacheControl
-							.maxAge(cachePeriod.toMillis(), TimeUnit.MILLISECONDS));
+					cacheControl.sMaxAge(cachePeriod.toMillis(), TimeUnit.MILLISECONDS);
```
Comment (bclozel) [inline spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration.java]: same typo

**Comment by ayudovin** [comment]
@bclozel, thank you for your comments, I have added changes, Can you see again ?

**Comment by bclozel** [comment]
Merged.
Thanks @ayudovin !

**Pull request closed without merging**